### ISSUE TITLE
[2.6] find-window behavior change

### ISSUE THREAD

**Issue by juancampa** [tmux/tmux#1129]
Just upgraded to 2.6.

When I search using `find-window -N '%%'` and there's only **one window that matches**, tmux 2.6 shows a tree view. The previous behavior was switching to the one matching window without showing the tree.

I see `find-window` it was rewritten in terms of the new tree (kudos on unifying this). Is there a plan to bring this behavior back? Is there any other way to quickly switch between windows without having to see a tree with only one window?

Thanks

## Environment
```
$ uname -sp && tmux -V && echo $TERM
Darwin i386
tmux 2.6
screen-256color
```

**Comment by nicm** [comment]
select-window?


On 22 Oct 2017 1:10 am, "Juan Campa" <notifications@github.com> wrote:

Just upgraded to 2.6.

When I search using find-window -N '%%' and there's only one window that
matches, tmux 2.6 now shows a tree view. The previous behavior was
switching to it without showing the tree.

I see it was rewritten in terms of the new tree (kudos on unifying this).
Is there a plan to bring this behavior back? Is there any other way to
quickly switch between windows without having to see a tree with only one
window?

Thanks
Environment

$ uname -sp && tmux -V && echo $TERM
Darwin i386
tmux 2.6
screen-256color

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/tmux/tmux/issues/1129>, or mute the thread

**Comment by juancampa** [comment]
Thanks for the prompt reply. I just tried `select-window` but there's no way to specify a search string like in `find-window`.

Basically I want to tell tmux: "switch to the window titled 'backend'"

**Comment by nicm** [comment]
If you really want title rather than name then you will need to do a script
that does something like

W=$(tmux lsw -F '#{pane_title} #{window_id}'|awk "/^$1/{print \$2}")
tmux selectw -t$W

If you do want name rather than title you can just use select-window...
look at -t.



On 22 Oct 2017 1:33 am, "Juan Campa" <notifications@github.com> wrote:

Thanks for the prompt reply. I just tried select-window but there's no way
to specify a search string like in find-window.

Basically I want to tell tmux: "switch to the window titled 'backend'"

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/tmux/tmux/issues/1129#issuecomment-338442072>, or mute
the thread

**Comment by juancampa** [comment]
Whoa thanks for coming up with the script. I just tinkered around with `-t` and realized that it's what I needed. When I first looked I assumed that it only accepted the window index.
